Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsZipPreprocessExecuteAndRunUnitTestsWithDOMAndMochaExecutionStrategy.cs

```diff
@@ -0,0 +1,322 @@
+        }
+
+        ~NodeJsZipPreprocessExecuteAndRunUnitTestsWithDomAndMochaExecutionStrategy()
+        {
+            DirectoryHelpers.SafeDeleteDirectory(this.WorkingDirectory, true);
+        }
+
+        protected string BrowserifyModulePath { get; }
+
+        protected string BabelifyModulePath { get; }
+
+        protected string EcmaScriptImportPluginPath { get; }
+
+        protected string WorkingDirectory { get; set; }
+
+        protected string ProgramEntryPath { get; set; }
+
+        protected override string JsCodeRequiredModules => base.JsCodeRequiredModules + @",
+    browserify = require('" + this.BrowserifyModulePath + @"'),
+    streamJs = require('stream'),
+    stream = new streamJs.PassThrough();";
+
+        protected string JsNodeDisableCode => @"
+DataView = undefined;
+DTRACE_NET_SERVER_CONNECTION = undefined;
+// DTRACE_NET_STREAM_END = undefined;
+DTRACE_NET_SOCKET_READ = undefined;
+DTRACE_NET_SOCKET_WRITE = undefined;
+DTRACE_HTTP_SERVER_REQUEST = undefined;
+DTRACE_HTTP_SERVER_RESPONSE = undefined;
+DTRACE_HTTP_CLIENT_REQUEST = undefined;
+DTRACE_HTTP_CLIENT_RESPONSE = undefined;
+COUNTER_NET_SERVER_CONNECTION = undefined;
+COUNTER_NET_SERVER_CONNECTION_CLOSE = undefined;
+COUNTER_HTTP_SERVER_REQUEST = undefined;
+COUNTER_HTTP_SERVER_RESPONSE = undefined;
+COUNTER_HTTP_CLIENT_REQUEST = undefined;
+COUNTER_HTTP_CLIENT_RESPONSE = undefined;
+process.argv = undefined;
+process.versions = undefined;
+process.env = { NODE_DEBUG: false };
+process.addListener = undefined;
+process.EventEmitter = undefined;
+process.mainModule = undefined;
+process.config = undefined;
+// process.on = undefined;
+process.openStdin = undefined;
+process.chdir = undefined;
+process.cwd = undefined;
+process.exit = undefined;
+process.umask = undefined;
+// GLOBAL = undefined;
+// root = undefined;
+// global = {};
+setInterval = undefined;
+// clearTimeout = undefined;
+clearInterval = undefined;
+setImmediate = undefined;
+clearImmediate = undefined;
+module = undefined;
+require = undefined;
+msg = undefined;
+
+delete DataView;
+delete DTRACE_NET_SERVER_CONNECTION;
+// delete DTRACE_NET_STREAM_END;
+delete DTRACE_NET_SOCKET_READ;
+delete DTRACE_NET_SOCKET_WRITE;
+delete DTRACE_HTTP_SERVER_REQUEST;
+delete DTRACE_HTTP_SERVER_RESPONSE;
+delete DTRACE_HTTP_CLIENT_REQUEST;
+delete DTRACE_HTTP_CLIENT_RESPONSE;
+delete COUNTER_NET_SERVER_CONNECTION;
+delete COUNTER_NET_SERVER_CONNECTION_CLOSE;
+delete COUNTER_HTTP_SERVER_REQUEST;
+delete COUNTER_HTTP_SERVER_RESPONSE;
+delete COUNTER_HTTP_CLIENT_REQUEST;
+delete COUNTER_HTTP_CLIENT_RESPONSE;
+delete process.argv;
+delete process.exit;
+delete process.versions;
+// delete GLOBAL;
+// delete root;
+delete setInterval;
+// delete clearTimeout;
+delete clearInterval;
+delete setImmediate;
+delete clearImmediate;
+delete module;
+delete require;
+delete msg;
+
+process.exit = function () {};";
+
+        protected override string JsCodeTemplate => this.JsCodeRequiredModules +
```

Comment:
`this.JsCodeRequiredModules +` must be on the next line, indented with **one** tab.